### ISSUE TITLE
0.7 - Breaking changes

### ISSUE THREAD

**Issue by rrag** [rrag/react-stockcharts#206]
The issue description will be kept up to date reflecting the status and changes

- [x] In order to get the benefits of tree-shaking, and as a first step to splitting into different modules, all the individual components have to be imported by itself

```diff
-import { ChartCanvas, Chart, series, scale, coordinates, tooltip, axes, indicator, helper } from "react-stockcharts";
+import { ChartCanvas, Chart } from "react-stockcharts";

-var { CandlestickSeries, BarSeries, LineSeries, AreaSeries } = series;
+import {
+       CandlestickSeries,
+       BarSeries,
+       LineSeries,
+       AreaSeries,
+} from "react-stockcharts/lib/series";

-var { discontinuousTimeScaleProvider } = scale;
+import { discontinuousTimeScaleProvider } from "react-stockcharts/lib/scale";

-var { CrossHairCursor, MouseCoordinateX, MouseCoordinateY, CurrentCoordinate } = coordinates;
+import {
+       CrossHairCursor,
+       CurrentCoordinate,
+       MouseCoordinateX,
+       MouseCoordinateY,
+} from "react-stockcharts/lib/coordinates";

-var { OHLCTooltip, MovingAverageTooltip } = tooltip;
+import {
+       OHLCTooltip,
+       MovingAverageTooltip,
+} from "react-stockcharts/lib/tooltip";

-var { XAxis, YAxis } = axes;
+import { XAxis, YAxis } from "react-stockcharts/lib/axes";

-var { ema, sma, wma, tma } = indicator;
+import { ema, wma, sma, tma } from "react-stockcharts/lib/indicator";

-var { fitWidth } = helper;
+import { fitWidth } from "react-stockcharts/lib/helper";
```

- [x] Remove the usage of `calculator`, from all `XXXSeries` and `XXXTooltip` and from `ChartCanvas` what this means is that react-stockcharts will not calculate any indicators for you, it has to be done in userland *before* rendering the chart. There have been many questions about custom indicators and how to use indicators which are calculated on the server. After using the app myself, I think it is more intuitive to provide helpers to calculate indicators, so users can use them if needed (or use their own calculators) and provide `data[]` with the precalculated values
- [x] Remove `xScaleProvider` from `ChartCanvas` so for discontinuous scales, there will be a helper function provided which users can use to create a new `xScale` for discontinuous data and provide that as an input in `xScale`
- [x] Remove the contents of folder `lib/indicator` and move `lib/indicator/algorithm/` into `lib/indicator`
- [x] Investigate Interactive indicators to use canvas (again)
    Made some good progress on canvas interactives, you can find the work in progress [here](http://localhost:8080/documentation.html#/trendline) and [here](http://rrag.github.io/react-stockcharts-next/documentation.html#/fibonacci_retracement). New features hover, blur, click, drag events for the components. After drawing, click the component to select it and then drag it around. Different components can have different degrees of freedom, see the fib retracement e.g. where the top and bottom retracements move north-south, the edge circles move east-west and the rest of the lines move in all directions
- [ ] Interactive indicators to be touch friendly - not doing in 0.7
- [ ] In order to fix https://github.com/rrag/react-stockcharts/issues/165 I see no alternative other than to drop svg support, and use html `div/span` for tooltip text - this has been discussed earlier and I am still not certain what I want to do. SVG does not have a float based layout which is necessary to fix #165  - not doing in 0.7

Development of these are happening on the `next` branch so please review that before you want to submit any PR. Ideally I want to accept PRs only on the `next` branch unless it is for bugfixes to `0.6` without simple workarounds

**Comment by itsjimbo** [comment]
I know it is still premature to ask, but will this fall in line with react-fiber and take advantage of the improvements?  (They claim 100% backward compatibility, but you know how these things go...)
Maybe that should be added to the 0.7 release?

http://isfiberreadyyet.com/

**Comment by rrag** [comment]
I have not looked at any fiber specific improvements. Still going to rely on the existing api only. I also do not expect any performance improvements because of fiber.

There are may be 2 improvements I can think of which can improve perf, and they are both not on react. This is because canvas drawing happens without using any react `setState`

1. now every component does a
```
ctx.setTransform(1, 0, 0, 1, 0, 0);
ctx.scale(ratio, ratio);
ctx.translate(canvasOriginX, canvasOriginY);

// render the component

ctx.restore();
```
if there are say 10 components, then these transform, scale, translate, restore  dance happens 10 times. I dont know if these particularly cause any performance slowness, but that is the only extra work that happens.

2. there are now as of `0.7` [4 canvases](https://github.com/rrag/react-stockcharts/blob/next/src/lib/CanvasContainer.jsx#L33-L40), if we can remove the `bg` and cause a breaking change by removing the `Label` feature it could reduce some work that happens

Just curious, anything specific you had in mind which we can use in fiber?

**Comment by rrag** [comment]
Huge perf improvement in `next` version
upto 30 fps when panning lots of data at 2 points per pixel

https://www.youtube.com/watch?v=wWbE4KDRggQ

Check it out: http://rrag.github.io/fullpage/documentation.html

Appreciate feedback
cc @itsjimbo @WaiSiuKei

**Comment by madhumurugan89** [comment]
Hi, Please help me to find the examples for 0.7 version.

**Comment by theorye** [comment]
@madhumurugan89 They are in the next branch

**Comment by WaiSiuKei** [comment]
Hi @rrag , sorry for late reply, I am making something to demonstrate my opinion: 
You can extract the indicator calculation part as a standalone module, such as my [Finscript](https://waisiukei.github.io/finscript/)

**Comment by rrag** [comment]
@WaiSiuKei this is amazing. any plans of releasing this opensource?

**Comment by WaiSiuKei** [comment]
@rrag sourcecode is updated last night (UTC+8 Timezone)

**Comment by arash3tm** [comment]
@rrag will you add logatrithmic scale Axis in next changes?

**Comment by rrag** [comment]
@arash3tm you can already use log scales, like `<Chart yScale={d3.scaleLog()}`, default is a `scaleLinear` https://github.com/rrag/react-stockcharts/blob/master/src/lib/Chart.js#L98

**Comment by rrag** [comment]
Published 0.7

**Comment by congaductq** [comment]
Hi rrag,

How can I use log 10 scale? I've tried to pass `scaleLinear().base(10).domain([0.01, 100])` but it's did not works.

Thank you!
